test/letters/actions/letters.unit.spec.js

```diff
@@ -21,6 +36,208 @@ const frontEndAddress = {
   state: 'secret'
 };
 
+// TO-DO: Determine if this is needed
+const addressResponse = {
+  data: {
+    attributes: {
+      address: {
+        type: 'DOMESTIC',
+        addressEffectiveDate: '1973-01-01T05:00:00.000+00:00',
+        addressOne: '140 Rock Creek Church Rd NW',
+        addressTwo: '',
+        addressThree: '',
+        city: 'Washington',
+        stateCode: 'DC',
+        zipCode: '20011',
+        zipSuffix: '1865'
+      },
+      controlInformation: {
+        canUpdate: true,
+        corpAvailIndicator: true,
+        corpRecFoundIndicator: true,
+        hasNoBdnPaymentsIndicator: true,
+        isCompetentIndicator: true,
+        indentityIndicator: true,
+        indexIndicator: true,
+        noFiduciaryAssignedIndicator: true,
+        notDeceasedIndicator: true
+      }
+    }
+  }
+};
+
+/**
+ * Setup() for each test requires stubbing global fetch() and setting userToken
+ * in global sessionStorage. Teardown() resets everything back to normal
+ */
+let oldFetch;
+let oldSessionStorage;
+const setup = () => {
+  oldSessionStorage = global.sessionStorage;
+  oldFetch = global.fetch;
+  global.sessionStorage = {
+    userToken: '123abc'
+  };
+  global.fetch = sinon.stub();
+  global.fetch.returns(Promise.resolve({
+    headers: { get: () => 'application/json' },
+    ok: true,
+    json: () => Promise.resolve({})
+  }));
+};
+const teardown = () => {
+  global.fetch = oldFetch;
+  global.sessionStorage = oldSessionStorage;
+};
+const getState = () => ({});
+
+// without redux mock store - these tests pass
+describe('saveAddress', () => {
+  beforeEach(setup);
+  afterEach(teardown);
+
+  it('dispatches SAVE_ADDRESS_PENDING first', (done) => {
+    const thunk = saveAddress(frontEndAddress);
+    let callCount = 0;
+    const dispatch = sinon.spy((action) => {
+      // keep track of which dispatch() call we're asserting against
+      callCount += 1;
+      const { type } = action;
+      if (callCount === 1) {
+        try {
+          expect(type).to.equal(SAVE_ADDRESS_PENDING);
+          done();
+        } catch (error) {
+          done(error);
```

Comment:
Is this necessary? I think we can just do an `expect()` then `done()`; if `expect()` throws an error, isn't it caught by mocha? I mean, this should work just as well, but I think we don't need to try / catch it.

I could be wrong, though. :man_shrugging: